src/FileLoader.php

```diff
@@ -0,0 +1,181 @@
+<?php
+namespace Michaels\Manager;
+
+use Michaels\Manager\Contracts\DecoderInterface;
+use Michaels\Manager\Decoders\JsonDecoder;
+use Michaels\Manager\Bags\FileBag;
+use Michaels\Manager\Exceptions\UnsupportedFilesException;
+use Michaels\Manager\Exceptions\BadFileDataException;
+use Exception;
+
+
+class FileLoader
+{
+
+    private $supportedMimeTypes = array();
+
+    private $fileBag = array();
+
+    private $decoders = array();
+
+    private $manager;
+
+    private $unsupportedFiles = array();
+
+    private $decodedData = array();
+
+    /**
+     * Constructor - gets a manager instance to work with and sets up default decoders.
+     *
+     * @param Manager $manager
+     */
+
+    public function __construct(Manager $manager)
+    {
+        $this->manager = $manager;
+
+        $this->addDecoder(new JsonDecoder());
+        //$this->addDecoder(new YamlDecoder());
+
+    }
+
+    /**
+     * Add a file decoder to the FileLoader
+     *
+     * @param DecoderInterface $decoder
+     */
+    public function addDecoder(DecoderInterface $decoder)
+    {
+        $mimeTypes = $decoder->getMimeType();
+        if (empty($this->supportedMimeTypes)){
+            $this->supportedMimeTypes = $mimeTypes;
+        }else{
+            array_merge($this->supportedMimeTypes, $mimeTypes);
+        }
+        foreach($mimeTypes as $type)
+        {
+            $this->decoders[$type] = &$decoder;
```

Comment:
Ah, I wasn't sure about that. I'll remove the reference. I was thinking, no matter how many mime types are defined, they should all still point to the same decoder object.

Scott